File: back/api/app/types/Rol.php
<?php

namespace app\types;

enum Rol: string {
    case ADMIN = "ADMIN";
    case USER = "USER";
    case GUEST = "GUEST";
    case IOT = "IOT";

    public function getValue(): string
    {
        return $this->value;
    }

    public function getName(): string
    {
        return $this->name;
    }

    public function equals(Rol $targetRol): bool
    {
        return $this->value === $targetRol->getValue();
    }

}

File: back-go/constants/rol.go
package constants

type Rol int

const (
	ADMIN Rol = iota
	USER
	IOT
)

func (r Rol) String() string {
	switch r {
	case ADMIN:
		return "ADMIN"
	case USER:
		return "USER"
	case IOT:
		return "IOT"
	}
}
